### PR TITLE
Report: include partial's css only once

### LINKED ISSUE #1578: Report: move all CSS to `head`
Apart from it not being valid HTML, it's hard to debug it as the style tags now appear through the whole `body`.

If you don't want to move everything to `report.css`, maybe a build step could take care of this instead.

### PR DESCRIPTION
**Pull request by ebidel**
R: all

Fixes #1578.

Each html partial defines its own `<style>`. While this is convenient, it ends up injecting a bunch of duplicated CSS into the final report:

screen shot 2017-02-06 at 9 41 38 pm

This PR creates .css files for each partial and injects the styles once, at the top of the page.

### CODE REVIEW

**Comment by XhmikosR** [comment]
@ebidel: shouldn't this fix #1578?

**Comment by ebidel** [comment]
@XhmikosR it should take care of that too.

**Comment by XhmikosR** [comment]
Cool, nice! Now if you added CSS minifying it would fix #1552, but perhaps you want to tackle that another time.

**Comment by ebidel** [comment]
Been playing around with https://github.com/kangax/html-minifier but it's a huge dependency.

**Comment by XhmikosR** [comment]
Yeah, unfortunately that grew up a lot.

You could use a css minifier only for starters. It's just that with html-minifier everything would be taken care by one root dependency.

**Comment by XhmikosR** [comment]
I mean, it's using clean-css, uglify-js for CSS and JS respectively. The other deps are for unicode characters and some other stuff.

Personally, I've been using html-minifier for years in all my projects.

File: lighthouse-core/formatters/partials/speedline.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.speedline-measures {
```
Comment (ebidel) [inline lighthouse-core/formatters/partials/speedline.css]: @paulirish I think you got rid of this in another PR?

File: lighthouse-core/report/templates/report-template.html [hunk 2]
```diff
@@ -25,19 +24,20 @@
...
+  <link rel="shortcut icon" href="https://raw.githubusercontent.com/GoogleChrome/lighthouse/master/assets/lh_logo_icon.png">
```
Comment (ebidel) [inline lighthouse-core/report/templates/report-template.html]: This shaves some bytes but also allows us to keep the icon up to date.

File: lighthouse-core/report/templates/report-template.html [hunk 3]
```diff
@@ -25,19 +24,20 @@
...
   <title>Lighthouse report: {{ url }}</title>
-  {{#each stylesheets }}
-    <style>{{{ this }}}</style>
```
Comment (ebidel) [inline lighthouse-core/report/templates/report-template.html]: it was silly to create separate `<style>` for each each sheet.

File: lighthouse-core/report/templates/report-template.html (hunk 2 above)
Comment (XhmikosR) [inline lighthouse-core/report/templates/report-template.html]: Personally, I don't like that. You are loading a 1303x1303 image for what? For 48x48 max.

Comment (ebidel) [inline lighthouse-core/report/templates/report-template.html]: In theory, the 20kb will be cached and more shareable by the browser across different instances of the report.

Comment (XhmikosR) [inline lighthouse-core/report/templates/report-template.html]: Or... you can just link to viewer's 48x48 icon.

File: lighthouse-core/formatters/partials/speedline.css (hunk 1 above)
Comment (brendankenny) [inline lighthouse-core/formatters/partials/speedline.css]: #1650

Comment (brendankenny) [inline lighthouse-core/formatters/partials/speedline.css]: looks like you can just no include a css file for it (or leave an empty one so there's a place for future css?)

Comment (ebidel) [inline lighthouse-core/formatters/partials/speedline.css]: Removed the file. There are other partials that don't have a .css file.

Comment (ebidel) [inline lighthouse-core/formatters/partials/speedline.css]: Done

File: lighthouse-core/report/templates/report-template.html (hunk 2 above)
Comment (brendankenny) [inline lighthouse-core/report/templates/report-template.html]: it's just the favicon, so not sure if it matters, but this means the report won't be self contained

File: lighthouse-core/report/report-generator.js [hunk 4]
```diff
@@ -214,7 +214,20 @@ class ReportGenerator {
    * @return {!Array<string>} an array of CSS
    */
   getReportCSS() {
-    return [fs.readFileSync(path.join(__dirname, './styles/report.css'), 'utf8')];
+    // Cannot DRY this up and dynamically create paths because fs.readdirSync
+    // doesn't browserify well with a variable path. See https://github.com/substack/brfs/issues/36.
+    const partialStyles = [
+      fs.readFileSync(path.join(__dirname, '../formatters/partials',
```
Comment (brendankenny) [inline lighthouse-core/report/report-generator.js]: nit: can just do `__dirname + '../formatters/partials/critical-request-chains.css'` (for all of these)

File: lighthouse-core/report/templates/report-template.html [hunk 5]
```diff
@@ -25,19 +24,20 @@
...
-  {{/each}}
+  <style>
+    {{#each stylesheets }}
+      {{{ this }}}
```
Comment (paulirish) [inline lighthouse-core/report/templates/report-template.html]: can you add this for better debugging and editing w/ devtools?

```css
 /*# sourceURL=inline-styles-{{@index}}.css */
```

File: lighthouse-core/report/report-generator.js (hunk 4 above)
Comment (ebidel) [inline lighthouse-core/report/report-generator.js]: done

File: lighthouse-core/report/templates/report-template.html (hunk 5 above)
Comment (ebidel) [inline lighthouse-core/report/templates/report-template.html]: added for the single `<style>`

**Comment by paulirish** [comment]
Yup lgtm 
image

**Pull request merged**